File: render.py
#!/usr/bin/env python3
"""
Render cocktail-menu.html → output/cocktail-menu.pdf
using headless Chromium via Playwright (Node.js).
"""
import os, subprocess, sys

ROOT     = os.path.dirname(os.path.abspath(__file__))
HTML     = os.path.join(ROOT, 'cocktail-menu.html')
SCRIPT   = os.path.join(ROOT, 'render.js')
OUT_PDF  = os.path.join(ROOT, 'output', 'cocktail-menu.pdf')
NODE     = '/opt/node22/bin/node'

if not os.path.exists(HTML):
    sys.exit(f'ERROR: {HTML} not found. Run generate_menu.py first.')

result = subprocess.run([NODE, SCRIPT], capture_output=False, text=True)
if result.returncode != 0:
    sys.exit(f'Render failed (exit {result.returncode})')

size_kb = os.path.getsize(OUT_PDF) / 1024
print(f'\nPDF ready: {OUT_PDF}  ({size_kb:.0f} KB)')

File: render.js
// Render cocktail-menu.html to PDF using Playwright (Node.js)
const { chromium } = require('/opt/node22/lib/node_modules/playwright');
const path = require('path');
const fs = require('fs');

(async () => {
  const htmlPath = path.resolve(__dirname, 'cocktail-menu.html');
  const outDir = path.resolve(__dirname, 'output');
  const outPath = path.join(outDir, 'cocktail-menu.pdf');

  if (!fs.existsSync(outDir)) fs.mkdirSync(outDir, { recursive: true });

  console.log('Launching Chromium…');
  const browser = await chromium.launch({
    executablePath: '/opt/pw-browsers/chromium-1194/chrome-linux/chrome',
    args: ['--no-sandbox', '--disable-setuid-sandbox'],
  });

  const page = await browser.newPage();

  console.log('Loading HTML…');
  await page.goto('file://' + htmlPath, { waitUntil: 'networkidle', timeout: 30000 });

  // Give fonts a moment to fully render
  await page.waitForTimeout(1000);

  console.log('Generating PDF…');
  await page.pdf({
    path: outPath,
    width: '148mm',
    height: '210mm',
    printBackground: true,
    margin: { top: '0', right: '0', bottom: '0', left: '0' },
  });

  await browser.close();
  console.log(`PDF written: ${outPath}`);
  console.log(`Size: ${(fs.statSync(outPath).size / 1024).toFixed(0)} KB`);
})();
